### PR TITLE
Handle case when connection we're identifying gets closed

### PR DESCRIPTION
**Pull request by Vagabond**
This simply adds a try/catch

### CODE REVIEW

File: src/libp2p_transport_tcp.erl [hunk 1]
```diff
@@ -387,66 +387,72 @@ handle_info({handle_identify, Session, {error, Error}}, State=#state{}) ->
...
+                        false ->
+                            lager:debug("identify response with local address ~p that is not a listen addr socket ~p, ignoring",
+                                        [LocalAddr, NewListenAddrs]),
+                            %% this is likely a discovery session we dialed with unique_port
+                            %% we can't trust this for the purposes of the observed address
+                            {noreply, State}
+                    end
             end
+    catch
+        _:_ ->
```
Comment (evanmcc) [inline src/libp2p_transport_tcp.erl]: should we be logging things here so we can more selectively filter later?

Comment (madninja) [reply src/libp2p_transport_tcp.erl]: or maybe a (case catch) to keep it in the case statements and reduce one indent level in that insanely large code block?

Comment (madninja) [reply src/libp2p_transport_tcp.erl]: I realize there are a couple of places an error can be thrown so the top level try block may be the best way to do it, but this section of code is getting pretty unwieldy, no?

**Comment by Vagabond** [comment]
Comments should be addressed?

**Pull request merged**